hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/schemas/V0490FeeSchema.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.fees.schemas;
+
+import com.hedera.hapi.node.base.SemanticVersion;
+import com.hedera.hapi.node.base.TimestampSeconds;
+import com.hedera.hapi.node.transaction.ExchangeRate;
+import com.hedera.hapi.node.transaction.ExchangeRateSet;
+import com.hedera.node.app.service.mono.state.submerkle.ExchangeRates;
+import com.hedera.node.app.spi.state.MigrationContext;
+import com.hedera.node.app.spi.state.Schema;
+import com.hedera.node.app.spi.state.StateDefinition;
+import com.hedera.node.config.data.BootstrapConfig;
+import com.swirlds.platform.state.spi.WritableSingletonStateBase;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class V0490FeeSchema extends Schema {
+    public static final String MIDNIGHT_RATES_STATE_KEY = "MIDNIGHT_RATES";
+    private static final Logger log = LogManager.getLogger(V0490FeeSchema.class);
+
+    private static ExchangeRates fs;
+
+    /**
+     * The version of the schema.
+     */
+    private static final SemanticVersion VERSION =
+            SemanticVersion.newBuilder().major(0).minor(49).patch(0).build();
```

Comment:
nit : we can probably add a public static field for 0_49_0_VERSION

Reply:
I think we should keep SPI as Hedera-agnostic as possible; so e.g. Hedera release versions don't make sense there. 🤔